meta-facebook/at-mc/src/ipmi/plat_ipmi.c

```diff
@@ -179,9 +181,15 @@ int pal_write_read_cxl_fru(uint8_t optional, uint8_t fru_id, EEPROM_ENTRY *fru_e
 	}
 
 	struct k_mutex *mutex = get_i2c_mux_mutex(cxl_mux.bus);
-	int mutex_status = k_mutex_lock(mutex, K_MSEC(MUTEX_LOCK_INTERVAL_MS));
-	if (mutex_status != 0) {
-		LOG_ERR("Mutex lock fail, status: %d", mutex_status);
+	for (retry = 0; retry < 5; ++retry) {
+		mutex_status = k_mutex_lock(mutex, K_MSEC(MUTEX_LOCK_INTERVAL_MS));
```

Comment:
Why do we need a retry here?  Can't we just make the length of the lock attempt to be 5x longer?

Reply:
Hi @williamspatrick ,
We have tested that after locking for one second and retrying five times, the fru write/read stress will pass 500 times, but if it is locked for five seconds, the failure rate of writing fru is 20%.
We will study the part where these two methods have different test results.

Regards,
Lora.

Reply:
Hi @williamspatrick .
Because this commit contains a part (Unlock mutex when initial CXL information fails, avoiding it permanent occupation) that needs to be fixed for this release, if you are concerned about the retry practice of writing to CXL fru, do we need to re-create a commit to fix the problem with mutex not unlocked ?

Regards,
Lora.